### PR TITLE
fix: add app.mockLog() to improve app.expectLog() more stable

### PR DESCRIPTION
**Pull request by fengmk2**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,36 +25,36 @@
...
-    "detect-port": "^1.2.1",
-    "egg-utils": "^2.2.0",
+    "coffee": "^5.1.0",
+    "debug": "^4.0.1",
+    "detect-port": "^1.2.3",
+    "egg-utils": "^2.4.1",
     "extend2": "^1.0.0",
     "get-ready": "^2.0.1",
-    "globby": "^6.1.0",
+    "globby": "^8.0.1",
```
Comment (fengmk2) [inline package.json]: 看了一下这个是可以升级的，改成了 fast glob https://github.com/sindresorhus/globby/commits/master

**Comment by fengmk2** [comment]
@atian25 更多是应用层的测试。

File: app/extend/application.js [hunk 2]
```diff
@@ -308,7 +308,32 @@ module.exports = {
...
+   */
+  mockLog(logger) {
+    logger = logger || this.logger;
+    if (typeof logger === 'string') {
+      logger = this.getLogger(logger);
+    }
+    // make sure mock once
+    if (logger._mockLogs) return;
+
+    const Transport = require('egg-logger').Transport;
```
Comment (dead-horse) [inline app/extend/application.js]: 没有 require `egg-logger`，require 一下？

Comment (fengmk2) [reply app/extend/application.js]: egg-logger 嗯，我加上。

**Pull request merged**